### ISSUE TITLE
`navigation_and_routing` to handle initial path before login

### ISSUE THREAD

**Issue by asashour** [flutter/samples#947]
Would the below scenario add a value to the sample:

1. The user opens the web browser with `/book/3`
2. The web app redirects to `/signin`
3. Sign in

Expected
- The app redirects to `/book/3`

Actual
- The app redirects to `/books/popular`

**Comment by ramanjitsingh-hub** [comment]
I want to work on this problem can i be assigned to this problem

**Comment by RedBrogdon** [comment]
@johnpryan, since this was your sample originally, I'd love to have your opinion on this.

**Comment by johnpryan** [comment]
Makes sense, the way I would do this is to add a `redirect` query parameter when the app navigations to `/signin`. So instead of `/signin` this would be `/signin?redirect=/books/3`.

**Comment by RedBrogdon** [comment]
It looks like there may be two people interested in implementing this.

@asashour, you were the first to suggest the change, so if you're interested in implementing it, I'll leave it to you.

**Comment by asashour** [comment]
Thanks all for your feedback.

@ramanjitsingh-hub please go ahead, I hope one week is enough to raise a PR.

If you want more ideas, we can have an error message when the username/password is empty, or tests to be added for the login or the navigation. As mentioned in [CONTRIBUTING](https://github.com/flutter/samples/blob/master/CONTRIBUTING.md), it is better to discuss the changes before spending time implementing them.

**Comment by arteevraina** [comment]
Hi @RedBrogdon Is this issue still there in the app? I am not able to reproduce the scenario mentioned by @asashour .

**Comment by asashour** [comment]
The issue is still reprodubile.

The initial URL should be something like `http://localhost:56422/#/book/3`.

**Comment by agelospanagiotakis** [comment]
so is there any way to pass a redirect parameter ?
